Source/WebCore/page/Page.cpp

```diff
@@ -809,10 +809,11 @@ bool Page::findString(const String& target, FindOptions options, DidWrap* didWra
 }
 
 #if ENABLE(IMAGE_ANALYSIS)
-void Page::analyzeImagesForFindInPage()
+void Page::setImageQueueEmptyHysteresisThenAnalyzeImagesForFindInPage(Function<void()>&& callback)
 {
     if (settings().imageAnalysisDuringFindInPageEnabled()) {
-        if (RefPtr mainDocument = m_mainFrame->document(); mainDocument && !m_imageAnalysisQueue)
+        imageAnalysisQueue().setImageQueueEmptyHysteresis(WTFMove(callback));
+        if (RefPtr mainDocument = m_mainFrame->document(); mainDocument && !imageAnalysisQueue().analysisOfAllImagesOnPageHasStarted())
             imageAnalysisQueue().enqueueAllImages(*mainDocument, { }, { });
```

Comment:
I think it would be cleaner to rename this method to `enqueueAllImagesIfNeeded()`, and move the `m_analysisOfAllImagesOnPageHasStarted` check into `enqueueAllImagesIfNeeded()`. That way, each call site doesn't need to individually consult `analysisOfAllImagesOnPageHasStarted()`.